packages/bsky/src/services/indexing/index.ts

```diff
@@ -49,28 +45,59 @@ export class IndexingService {
   ) {
     this.db.assertTransaction()
     const indexer = this.findIndexerForCollection(uri.collection)
+    if (!indexer) return
     // @TODO(bsky) direct notifs
     const notifs =
       action === WriteOpAction.Create
         ? await indexer.insertRecord(uri, cid, obj, timestamp)
         : await indexer.updateRecord(uri, cid, obj, timestamp)
+    return notifs
   }
 
   async deleteRecord(uri: AtUri, cascading = false) {
     this.db.assertTransaction()
     const indexer = this.findIndexerForCollection(uri.collection)
+    if (!indexer) return
     // @TODO(bsky) direct notifs
     const notifs = await indexer.deleteRecord(uri, cascading)
+    return notifs
+  }
+
+  async indexActor(did: string, timestamp: string) {
+    const actor = await this.db.db
+      .selectFrom('actor')
+      .where('did', '=', did)
+      .selectAll()
+      .executeTakeFirst()
+    if (actor) {
+      return // @TODO deal with handle updates
+    }
+    const { pds, handle } = await this.didResolver.resolveAtpData(did)
+    const handleToDid = await resolveExternalHandle(pds, handle)
+    if (did !== handleToDid) {
+      return // No bidirectional link between did and handle
+    }
```

Comment:
@dholms wanted to double-check, does this seem like the right approach for indexing the actor's handle?

Reply:
yup yup that seems good 👍 

I guess just some worries about retry logic there with DNS propagation? 

We oughta get in those authoritative name server lookups soon 😅